frontend/src/components/QueryBox.js

```diff
@@ -1,20 +1,60 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { Row, Col, Button } from 'react-bootstrap';
+import { Controlled as CodeMirror } from 'react-codemirror2';
+
+import 'codemirror/lib/codemirror.css';
+import 'codemirror/mode/sql/sql';
+import 'codemirror/addon/display/placeholder';
+import 'codemirror/addon/edit/matchbrackets';
+import 'codemirror/addon/hint/show-hint.css';
+import 'codemirror/addon/hint/show-hint';
+import 'codemirror/addon/hint/sql-hint';
+
 import './QueryBox.less';
 
 class QueryBox extends Component {
+  schemaToCodeMirror(schema) {
+    return schema.reduce(
+      (prevVal, currVal) => ({
+        ...prevVal,
+        [currVal.table]: currVal.columns.map(col => col.name)
+      }),
+      {}
+    );
+  }
+
   render() {
+    const tables = this.props.schema
+      ? this.schemaToCodeMirror(this.props.schema)
+      : {};
+
+    const options = {
```

Comment:
Isn't really important, but I wouldn't convert schema and create options object on every render (means every key press). Consider to create the codemirror configuration in constructor and update it on componentWillReceiveProps when schema or `handleSubmit` are changed.

Reply:
You are right, fixed in d2fb518

Reply:
rebased, now it's fixed in https://github.com/src-d/gitbase-playground/pull/35/commits/ce8f1dd8d7649dd3a4b55e822dfc0f7453ac6506